zun_kata.md

```diff
@@ -0,0 +1,148 @@
+# Introduction
+
+This guide describes how to get Kata containers working with OpenStack Zun using DevStack on Ubuntu 16.04.  It is based on the following links at the time of writing this guide.  Running DevStack with this guide will setup Docker and Clear Containers 2.0 which you will replace with Kata Containers.  
+https://docs.openstack.org/zun/latest/contributor/quickstart.html
+https://docs.openstack.org/zun/latest/admin/clear-containers.html
+https://github.com/clearcontainers/runtime/wiki/Running-Kata-Containers-with-the-Clear-Containers-runtime
+
+
+# Get go 1.8.3:
+
+```
+wget https://storage.googleapis.com/golang/go1.8.3.linux-amd64.tar.gz
+sudo tar -xvf go1.8.3.linux-amd64.tar.gz -C /usr/local/
+mkdir -p $HOME/go/src
+export GOPATH=$HOME/go
+export PATH=$PATH:/usr/local/go/bin:$GOPATH/bin
+go version
+```
+
+# Setup build tools
+
+```
+sudo apt install build-essential
+sudo apt install git
+```
+
+# Setup OpenStack DevStack
+
+```
+sudo mkdir -p /opt/stack
+sudo chown $USER /opt/stack
+git clone https://git.openstack.org/openstack-dev/devstack /opt/stack/devstack
+HOST_IP="$(ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1 -d'/')"
+git clone https://git.openstack.org/openstack/zun /opt/stack/zun
+cat /opt/stack/zun/devstack/local.conf.sample \
+    | sed "s/HOST_IP=.*/HOST_IP=$HOST_IP/" \
+    > /opt/stack/devstack/local.conf
+sed -i "s/KURYR_CAPABILITY_SCOPE=.*/KURYR_CAPABILITY_SCOPE=local/" /opt/stack/devstack/local.conf
+echo "ENABLE_CLEAR_CONTAINER=true" >> /opt/stack/devstack/local.conf
+/opt/stack/devstack/stack.sh
+source /opt/stack/devstack/openrc admin admin
+```
+
+This will start OpenStack DevStack with Zun support.  Test it out with runc to make sure everything installed correctly and is working.
+
+```
+zun run --name test cirros ping -c 4 8.8.8.8
+zun list
+zun logs test
+
+zun delete test
+```
+
+# Build and install the Kata proxy, Kata shim, and Kata container runtime and enable full debug
+
+```
+go get -d github.com/kata-containers/proxy
+cd $GOPATH/src/github.com/kata-containers/proxy && make && sudo make install
+go get -d github.com/kata-containers/shim
+cd $GOPATH/src/github.com/kata-containers/shim && make && sudo make install
+go get -d github.com/clearcontainers/runtime
+cd $GOPATH/src/github.com/clearcontainers/runtime
+make build-kata-system && sudo -E PATH=$PATH make install-kata-system
+sudo sed -i -e 's/^# *\(enable_debug\).*=.*$/\1 = true/g' /usr/share/defaults/kata-containers/configuration.toml
+```
+
+# Build the Kata agent and create a guest OS image. 
+
+Docker should be installed and runc should be the default runtime or else the image won't build. When DevStack was setup earlier it should have installed Docker and used runc as the default runtime. This will build the Kata agent, setup a guest OS image layout, copy the agent to the image, build the image, and copy the image to the host OS.  
+
+1. Build the Kata agent and create a rootfs image
```

Comment:
I suggest using a colon at the end of each list item. Currently steps 1 and 3 have no punctuation, and step 2 has a period.